taskcat/stacker.py

```diff
@@ -1288,24 +1284,10 @@ def check_yaml(self, yamlin, quite=None, strict=None):
             return False
         return True
 
-    def get_client(self, service_type, client_region):
-        if self._auth_mode is 'keys':
-            boto3.setup_default_session(
-                aws_access_key_id=self._aws_access_key,
-                aws_secret_access_key=self._aws_secret_key)
-            boto_client = boto3.client(service_type, client_region)
-            return boto_client
-        elif self._auth_mode is 'profile':
-            boto3.setup_default_session(profile_name=self._boto_profile)
-            boto_client = self._boto_client.get(service_type, client_region)
-            return boto_client
-        else:
-            boto_client = self._boto_client.get(service_type, client_region)
-            return boto_client
-
-
     # Set AWS Credentials
-    def aws_api_init(self, args):
+    def aws_api_init(self, auth_mode='role', aws_access_key=None,
```

Comment:
can we remove auth_mode parameter and just make a decision of the auth mode by considering which other parameters were passed in to this call?